packages/shared/lib/services/sync/orchestrator.service.ts

```diff
@@ -511,13 +517,42 @@ export class OrchestratorService {
             );
         }
     }
-    // TODO:
-    private async fetchSyncData(syncId: string, environmentId: number) {
-        const syncClient = await SyncClient.getInstance();
-        const schedule = await getSchedule(syncId);
-        const latestJob = await getLatestSyncJob(syncId);
-        let status = this.classifySyncStatus(latestJob?.status as SyncStatus, schedule?.status as ScheduleStatus);
 
+    private async syncStatus(sync: Sync, environmentId: number, includeJobStatus: boolean, orchestrator: Orchestrator): Promise<ReportedSyncJobStatus> {
+        const isGloballyEnabled = await featureFlags.isEnabled('orchestrator:schedule', 'global', false);
+        const isEnvEnabled = await featureFlags.isEnabled('orchestrator:schedule', `${environmentId}`, false);
+        const isOrchestrator = isGloballyEnabled || isEnvEnabled;
+        if (isOrchestrator) {
+            const latestJob = await getLatestSyncJob(sync.id);
+            const schedules = await orchestrator.searchSchedules([{ syncId: sync.id, environmentId }]);
+            if (schedules.isErr()) {
+                throw new Error(`Failed to get schedule for sync ${sync.id} in environment ${environmentId}: ${schedules.error}`);
```

Comment:
is error an Error object? if yes, needs a stringification

Reply:
decf602c